src/generators.rs

```diff
@@ -983,6 +983,139 @@ pub fn directed_grid_graph(
     })
 }
 
+/// Generate an undirected heavy hex graph.
```

Comment:
A useful reference here is Fig 2 of https://arxiv.org/pdf/1907.09528.pdf , and here's some ASCII art:

```
... D-S-D   D ...
    |   |   |
...-F   F-S-F ...
    |   |   |
... D   D   D ...
    |   |   |
... F-S-F   F-...
    |   |   |
    .........
    |   |   |
... D   D   D ...
    |   |   |
...-F   F-S-F ...
    |   |   |
... D   D   D ...
    |   |   |
... F-S-F   F-...
    |   |   |
    .........
    |   |   |
... D   D   D ...
    |   |   |
...-F   F-S-F ...
    |   |   |
... D   D   D ...
    |   |   |
... F-S-F   F-...
    |   |   |
... D   D-S-D ...
```